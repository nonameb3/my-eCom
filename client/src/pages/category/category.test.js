import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configStore from "redux-mock-store";

import Category from "./category.component";
import CollectionItem from "../../components/collection-item/collection-item.component";

const mockStore = configStore();

let wrapper;
const mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }];
const props = { match: { params: { categoryId: "test" } } };
const store = mockStore({
  shop: {
    collection: {
      test: { items: mockItems, title: "test" }
    }
  }
});

describe("Category component", () => {
  it("should render catagory compoent", () => {
    wrapper = shallow(
      <Provider store={store}>
        <Category {...props} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should rander collection-items same as number of collection arry", () => {
    wrapper = mount(
      <Provider store={store}>
        <Category {...props} />
      </Provider>
    );
    expect(wrapper.find(CollectionItem).length).toBe(mockItems.length);
  });
});
