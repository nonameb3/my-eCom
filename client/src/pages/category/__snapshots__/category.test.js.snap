// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category component should render catagory compoent 1`] = `ShallowWrapper {}`;
